### PR TITLE
feat(python): enhance `selectors` expansion function, so it can operate on a schema as well as a frame

### PR DESCRIPTION
**Pull request by alexander-beedie**
Ref: https://github.com/pola-rs/polars/pull/10255#issuecomment-1667790606

* **New**: allows selector expansion function to take a bare schema dictionary.
* Deprecates the old name `selector_column_names` for the clearer `expand_selector`.

## Example

```python
import polars.selectors as cs
import polars as pl

df = pl.DataFrame({
    "foo": ["a", "a", "b", "b", "b"],
    "baz": [100, 200, 300, 400, 500],
    "bar": ["x", "x", "x", "y", "z"],
})
```
From frame -
```python
cs.expand_selector( df, cs.string() )
# ('foo', 'bar') 
```
From schema dict _(new)_ -
```python
schema = { "colx":pl.Utf8, "coly":pl.Int8, "colz":pl.Float64 }
cs.expand_selector( schema, cs.numeric() )
# ('coly', 'colz')
```

### CODE REVIEW

File: py-polars/polars/selectors.py [hunk 1]
```diff
@@ -31,98 +32,129 @@
...
-
+    if isinstance(target, Mapping):
+        from polars.dataframe import DataFrame
 
-def is_selector(obj: Any) -> bool:
-    """
-    Indicate whether the given object/expression is a selector.
+        target = DataFrame(schema=target)
+    else:
+        target = target.clear()
```
Comment (ritchie46) [inline py-polars/polars/selectors.py:88]: Why does it have to be cleared?

File: py-polars/polars/selectors.py [hunk 2]
```diff
@@ -31,98 +32,129 @@
...
     selector
         An arbitrary polars selector (or compound selector).
 
+    """
+    return expand_selector(target=frame, selector=selector)
+
+
+def _expand_selectors(
+    frame: DataFrame | LazyFrame, items: Any, *more_items: Any
+) -> list[Any]:
```
[the remark this answers is not in the document]
Comment (alexander-beedie) [reply py-polars/polars/selectors.py:119]: It returns expanded selector colnames, and whatever was passed-in that wasn't a selector. So, if you know you _don't_ want integer columns _except_ for an integer called "id" you can simply declare `df.select( "id", ~cs.integer() )`.

The `expand_selectors` function would receive the list and expand `~cs.integer()` in place while keeping "id". Could also be something like `df.select( pl.Utf8, cs.temporal() )`, etc. Consequently the typing of the return is a list of `Any`.

File: py-polars/polars/selectors.py (hunk 1 above)
Comment (alexander-beedie) [reply py-polars/polars/selectors.py]: Ah! It doesn't; that's a leftover from the previous iteration that I meant to ditch now I'm updating it - fixed 👍

File: py-polars/polars/selectors.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexander-beedie) [reply py-polars/polars/selectors.py:117]: Ah, scratch that - I was thinking of the _private_ function that does a bit more for use internally when I wrote the above. Yes, this one should just be `list[str]` 😅

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**